File: session 18/example2.py
# operators
# 1. Arithmetic Operators: +, - , *, /, **, %
# 2. Assignment Operators: -=, +=, *=, /=, %=, //= (divide - floor), 
#     bitwise (1,0): &= (and), != (or), ^= (xor), >>= (right-shift) , <<= (left-shift)
# no1 =1  , no2 = 2 - > no1 &= no2
# 001, 010 - & => 000

#  a     b      xor
# true  false   true
# false true    true
# false false   false
# true  true    false


# >>=, 3 => 011, 5
# 0 -> 011  => 001
# 0 -> 001 => 000 = (0)

# 5 => 101, 3
# 0 -> 101 = 010 (2)
# 0 -> 010 = 001 (1)
# 0 -> 001 = 000
a= 5
b= 1
a >>= b
print(a)

# <<=

# Comparison Operator: >, < >=, <= 
# Logical: and, or, not - truth table
# Bitwise Operator: &, |, ^, !, << , >>
# Membership Operator: in, not in
list = [4, 6]
print(4 in list)
print(4 not in list)

# Indentity Operator: is, is not
print("rhea" is "rhea")
no1 = 3
no2 = 5
print(no1 is no2)
print(no1 is not no2)

File: session 11/example2.js
//3. logical operators: 
// &&, ||, !

// a = true, !a = false
// a = false, !a = true
console.log("!true", !true);

// a = true and b = true
//  a       b           a && b      a || b
// true     true        true        true
// false    false       false       false
// true     false       false       true
// false    true        false       true
console.log("3 < 5 && 7 > 5", 3 < 5 && 7 > 5);
//  1 <= x <= 10
let x=5;
console.log("x >= 1 && x <= 10", x >= 1 && x <= 10);
//          true    &&  true = true
console.log("x >=1 || x <= 4", x >=1 || x <= 4);
//          true   || false => true

// console.log("enter your name");

//4. asignment operators:
// +, - , *, /, %, ** -> arithmetic operator
console.log("\n");
let b = 3;
console.log("B: ", b);
b = b + 9;
console.log("B: ", b);
b += 1; // b = b + 1;
console.log("B: ", b);

b -= 9;
console.log("B: ", b);
b /= 2;
console.log("B: ", b);

b **= 3; // (2)^ 3 = (2 * 2 * 2 = 8)
console.log("B: ", b);

b %= 3; // 8/3 = 2, rem =2
console.log("B: ", b);